python/lsst/ctrl/bps/wms/panda/idds_tasks.py

```diff
@@ -96,11 +96,15 @@ def define_tasks(self):
             task = RubinTask()
             task.step = task_name
             task.name = task.step
-            task.queue = self.computing_queue_himem if self.tasks_steps[task_name] \
-                in self.himem_tasks else self.computing_queue
+            bps_node = next(filter(lambda x: x['job'].label == self.tasks_steps[task_name],
+                                   self.bps_workflow.nodes.values()))['job']

```

Comment:
If I got it right it looks like, for _each_ task, you iterate over _all_ nodes of the generic workflow to find a node with a job having a specific label. Is there a reason why you can't create and use some kind of a "lookup table" (for example, a dictionary mapping `task_steps` keys onto these nodes)?

Reply:
Filter provides the iterator and the values are computed lazy. This is why the next function will lead to finding only a single bps workflow node per task. The lookup table could do the same, I agree, but will require an extra variable. I would leave as it is now, if it is OK with you. Thank you!